packages/shared/src/components/MainLayout.tsx

```diff
@@ -238,3 +239,5 @@ export default function MainLayout({
     </div>
   );
 }
+
+export default withFeaturesBoundary(MainLayout);
```

Comment:
This is pretty strict, means every MainLayout element will wait for features. Is this what we want?

Reply:
Yeah it is, that is why I added a note in description to we should strive using this as "lowest" in component tree as possible. 

That said, `MainLayout` waits for `isPageReady` which has `isAuthReady` which waits for first boot load, so this does not add any additional time for the user in our current setup but it does make sure features are loaded from the start.

Reply:
That would complicate it I think, while HOC can be used dynamically it would introduce again cases where based on props changes you could render first without HOC and then with HOC which would again possibly let app render without features -> which would beat the purpose of using HOC in the first place.